Source/WebCore/html/canvas/GPUCanvasContextCocoa.cpp

```diff
@@ -99,17 +106,22 @@ void GPUCanvasContextCocoa::configure(GPUCanvasConfiguration&& configuration)
     reshape(canvasSize.width(), canvasSize.height());
 }
 
-void GPUCanvasContextCocoa::createPresentationContextIfNeeded()
+void GPUCanvasContextCocoa::unconfigurePresentationContextIfNeeded()
 {
-    if (!m_configuration)
+    if (!m_presentationContextIsConfigured)
         return;
 
-    GPUPresentationContextDescriptor presentationContextDescriptor = {
-        // FIXME: https://bugs.webkit.org/show_bug.cgi?id=250955 Add integration with the compositor here.
-    };
+    m_presentationContext->unconfigure();
+    m_presentationContextIsConfigured = false;
+}
 
-    m_presentationContext = m_gpu->createPresentationContext(presentationContextDescriptor);
-    ASSERT(m_presentationContext);
+void GPUCanvasContextCocoa::configurePresentationContextIfNeeded()
+{
+    if (m_presentationContextIsConfigured)
+        return;
+
+    if (!m_configuration)
+        return;
```

Comment:
when requesting to configure and there is no configuration, is that an error?

Reply:
Yes, though I'm going to end up rewriting this function entirely (as in https://github.com/WebKit/WebKit/pull/8963) so I didn't want to spend a lot of time polishing up the existing code in this function.